ggml-qnn.cpp

```diff
@@ -0,0 +1,3336 @@
+                              const ggml_tensor * src0, const ggml_tensor * src1,
+                              ggml_tensor * dst) {
+}
+
+static void ggml_qnn_cpy(ggml_backend_qnn_context * ctx, const ggml_tensor * src0,
+                         const ggml_tensor * src1, ggml_tensor * dst) {
+}
+
+static void ggml_qnn_dup(ggml_backend_qnn_context * ctx, const ggml_tensor * src0,
+                         const ggml_tensor * src1, ggml_tensor * dst) {
+    ggml_qnn_cpy(ctx, src0, dst, nullptr);
+    (void) src1;
+}
+
+static void ggml_qnn_mul_mat_id(ggml_backend_qnn_context * ctx,
+                                const ggml_tensor * src0,
+                                const ggml_tensor * src1, ggml_tensor * dst) {
+}
+
+static void ggml_qnn_scale(ggml_backend_qnn_context * ctx,
+                           const ggml_tensor * src0, const ggml_tensor * src1,
+                           ggml_tensor * dst) {
+}
+
+static void ggml_qnn_clamp(ggml_backend_qnn_context * ctx,
+                           const ggml_tensor * src0, const ggml_tensor * src1,
+                           ggml_tensor * dst) {
+}
+
+static void ggml_qnn_diag_mask_inf(ggml_backend_qnn_context * ctx,
+                                   const ggml_tensor * src0,
+                                   const ggml_tensor * src1, ggml_tensor * dst) {
+}
+
+static void ggml_qnn_soft_max(ggml_backend_qnn_context * ctx,
+                              const ggml_tensor * src0, const ggml_tensor * src1,
+                              ggml_tensor * dst) {
+}
+
+static void ggml_qnn_rope(ggml_backend_qnn_context * ctx,
+                          const ggml_tensor * src0, const ggml_tensor * src1,
+                          ggml_tensor * dst) {
+    GGML_ASSERT(ggml_is_contiguous(src0));
+}
+
+static void ggml_qnn_pool2d(ggml_backend_qnn_context * ctx,
+                            const ggml_tensor * src0, const ggml_tensor * src1,
+                            ggml_tensor * dst) {
+}
+
+static void ggml_qnn_im2col(ggml_backend_qnn_context * ctx,
+                            const ggml_tensor * src0, const ggml_tensor * src1,
+                            ggml_tensor * dst) {
+}
+
+static void ggml_qnn_sum_rows(ggml_backend_qnn_context * ctx,
+                              const ggml_tensor * src0, const ggml_tensor * src1,
+                              ggml_tensor * dst) {
+    GGML_ASSERT(ggml_is_contiguous(src0));
+}
+
+static void ggml_qnn_argsort(ggml_backend_qnn_context * ctx,
+                             const ggml_tensor * src0, const ggml_tensor * src1,
+                             ggml_tensor * dst) {
+    GGML_ASSERT(ggml_is_contiguous(src0));
+}
+
+static void ggml_qnn_nop(ggml_backend_qnn_context * ctx, const ggml_tensor * src0,
+                         const ggml_tensor * src1, ggml_tensor * dst) {
+    (void)src0;
+    (void)src1;
+    (void)dst;
+}
+
+bool ggml_qnn_compute_forward(ggml_backend_qnn_context * ctx,
```

Comment:
from onnx, there's a qnn op list, maybe we can take it as an example to add more op here:
https://github.com/microsoft/onnxruntime/blob/043ef5c95fad9a19a5563103cd7929633468ab4f/onnxruntime/core/providers/qnn/builder/opbuilder/base_op_builder.h#L107

Reply:
thanks for your time and review. I personally think it's not essential currently.

Reply:
sorry for make myself not clear enough, since there're lots of ggml operator unimplemented in this backend, maybe we can check out onnx repo (looks they've utilized more qnn operators than we did here), and follow their source code to support more opearators for qnn backend

Reply:
I tried but I personally don't think there are much valuable references in onnx's QNN backend.

Reply:
hello, I submit a new commit accordingly and I hope this commit can solve your concern and it could be a Minimum Viable Pullrequst) style PR. pls take time to have a look. I'd like to mark this loop as solved if you agree that. thanks so much.